code/modules/clothing/clothing.dm

```diff
@@ -198,6 +200,29 @@ BLIND     // can't see anything
 		var/mob/M = loc
 		M.update_inv_shoes()
 
+/obj/item/clothing/shoes/attackby(obj/item/I, mob/user, params)
+	..()
+	if(!can_hold_knives)
+		return
+	if(held_knife)
+		user << "<span class='notice'>There's already something in [src].</span>"
+		return
+	if(istype(I, /obj/item/weapon/kitchen/knife) || istype(I, /obj/item/weapon/pen) || istype(I, /obj/item/weapon/switchblade))//can hold both regular pens and energy daggers. made for your every-day tactical librarians/murderers.

```

Comment:
it's not a var it's a hardcoded check
make it a var/list in the shoes and is_type_in_list I on it